src/i_sound.c

```diff
@@ -481,113 +510,26 @@ void I_UpdateSound(void)
 void I_ShutdownSound(void)
 {
     int i;
-    ALCcontext *context;
-    ALCdevice *device;
 
     if (!snd_init)
     {
         return;
     }
 
-    context = alcGetCurrentContext();
-
-    if (!context)
-    {
-       return;
-    }
-
-    device = alcGetContextsDevice(context);
+    sound_module->ShutdownSound();
 
-    alDeleteSources(MAX_CHANNELS, openal_sources);
     for (i = 0; i < num_sfx; ++i)
     {
         if (S_sfx[i].cached)
         {
-            alDeleteBuffers(1, &S_sfx[i].buffer);
+            S_sfx[i].buffer = 0;
```

Comment:
According to the OpenAL specification, `buffer == 0` may be a valid identifier. That's why I added a `cached` field to `sfxinfo_t`.

Reply:
Maybe that entire for loop can be deleted since we're shutting down.

Reply:
Also, perhaps we can remove the `CacheSound()` function in `i_sound.c` (the PC Speaker module doesn't need to cache the sound).

Reply:
Even though it's harmless to remove, I'll leave this alone for now so any changes can be part of a PR that addresses `sfxinfo_t` in the future.

Reply:
> Also, perhaps we can remove the CacheSound() function in i_sound.c (the PC Speaker module doesn't need to cache the sound).

I took a second look at this to see if I could prep things in advance for PC speaker support, but it's complicated with dsdhacked.c, lumpnum tracking, and so on. Everything that modifies `S_sfx` needs a review to make sure it doesn't cause problems when switching between "DP" and "DS" lumps while the game is running.

Also, for now I'll skip the in-game HRTF .mhr file switching because Debian/Ubuntu uses a very old version of OpenAL Soft with a v2 file format, while newer versions [starting with 1.21.0](https://github.com/kcat/openal-soft/commit/be89684f3ccc06f56b69bc5c1d54a217b54fcfa2) use v3. That'll cause issues since most .mhr files shared online are v3. Some kind of special handling or version checking is required. I think this is an important feature because there are good alternatives to the default HRTF that comes with OpenAL Soft, such as [SADIE KEMAR or KU100](https://www.york.ac.uk/sadie-project/database.html) which I converted to .mhr and evaluated. Google used SADIE for their VR audio which is a good indication of the quality.

Reply:
> I took a second look at this to see if I could prep things in advance for PC speaker support, but it's complicated with dsdhacked.c, lumpnum tracking, and so on. 

I think we can ignore the "DP" sounds in dehacked etc, I'm pretty sure no one has ever tried to replace them.

> Also, for now I'll skip the in-game HRTF .mhr file switching because Debian/Ubuntu uses a very old version of OpenAL Soft with a v2 file format...

Interestingly, the default settings seem to work well with my headphones. I tested the current OpenAL on Windows only.

Reply:
> Interestingly, the default settings seem to work well with my headphones. I tested the current OpenAL on Windows only.

If you have time, try these out, I thought they were pretty good alternatives:
[SADIE_KEMAR.zip](https://github.com/fabiangreffrath/woof/files/11932375/SADIE_KEMAR.zip)
[SADIE_KU100.zip](https://github.com/fabiangreffrath/woof/files/11932376/SADIE_KU100.zip)

Copy one file at a time to `%AppData%\openal\hrtf` on Windows or `$HOME/.local/share/openal/hrtf` on Linux (OpenAL Soft 1.21.0 or higher). Only the first file in the folder will be used. You can use environment variables `ALSOFT_LOGFILE` set to a log file and `ALSOFT_LOGLEVEL` set to `3` to see what OpenAL Soft is actually doing.

```
$ export ALSOFT_LOGFILE=~/alsoft.log
$ export ALSOFT_LOGLEVEL=3
$ ./woof
[...]
$ grep GetLoadedHrtf ~/alsoft.log
AL lib: (II) GetLoadedHrtf: Loading /usr/share/openal/hrtf/default-44100.mhr...
AL lib: (II) GetLoadedHrtf: Detected data set format v2
AL lib: (II) GetLoadedHrtf: Loaded HRTF support for format: Stereo 44100hz
```

Reply:
Thanks for the update!

> I think we can ignore the "DP" sounds in dehacked etc, I'm pretty sure no one has ever tried to replace them.

Anyway, whatever puts the PC Speaker emulation at risk will have to wait, as apparently that's the next big thing on the list.

> Interestingly, the default settings seem to work well with my headphones. I tested the current OpenAL on Windows only.

Fo me too, so I think a HRTF profile browser can also wait (and I expect most non-technical users to be sheer overburdened with that choice)! Meanwhile, I have updated the openal-soft package in Debian to 1.23.1, but the stable releases will have to live with the outdated 1.19.1 for quite some time.

Reply:
On Windows, OpenAL-Soft looks for `%AppData%\Roaming\openal\hrtf`, not `%AppData%\openal\hrtf`. I tried both SADIE profiles and for some reason the default profile sounds better to me.

@fabiangreffrath Should we do another bug fix release until we merge this?

Reply:
Okay, I think users who care enough to change the file can make their own alsoft.ini/.conf and tweak whatever they want to their preference.

Reply:
> @fabiangreffrath Should we do another bug fix release until we merge this?

Yes, good idea.

Reply:
We'll need a Changelog for that. 😉

Reply:
Sure, I can make a changelog. Not sure how to fix that weapon rendering bug.

Reply:
Not sure, either. Might be a stupid fixed_t rounding bug. 🤷